File: src/jp/mcbe/fuyutsuki/Texter/data/OldFloatingTextData.php
<?php

/**
 * // English
 *
 * Texter, the display FloatingTextPerticle plugin for PocketMine-MP
 * Copyright (c) 2019-2021 yuko fuyutsuki < https://github.com/fuyutsuki >
 *
 * This software is distributed under "NCSA license".
 * You should have received a copy of the NCSA license
 * along with this program.  If not, see
 * < https://opensource.org/licenses/NCSA >.
 *
 * ---------------------------------------------------------------------
 * // 日本語
 *
 * TexterはPocketMine-MP向けのFloatingTextPerticleを表示するプラグインです
 * Copyright (c) 2019-2021 yuko fuyutsuki < https://github.com/fuyutsuki >
 *
 * このソフトウェアは"NCSAライセンス"下で配布されています。
 * あなたはこのプログラムと共にNCSAライセンスのコピーを受け取ったはずです。
 * 受け取っていない場合、下記のURLからご覧ください。
 * < https://opensource.org/licenses/NCSA >
 */

declare(strict_types=1);

namespace jp\mcbe\fuyutsuki\Texter\data;

use jp\mcbe\fuyutsuki\Texter\text\FloatingTextCluster;
use pocketmine\plugin\PluginBase;
use pocketmine\utils\Config;

class OldFloatingTextData extends Config {

	public const FILE_FT = "ft.json";
	public const FILE_UFT = "uft.json";

	public function __construct(
		private PluginBase $plugin,
		string $path,
		string $file
	) {
		parent::__construct($path.$file, Config::JSON);
	}

	public function convert() {
		$fts = $this->getAll();
		foreach ($fts as $worldName => $texts) {
			$floatingTextData = FloatingTextData::getInstance($worldName);
			if ($floatingTextData === null) {
				$floatingTextData = new FloatingTextData($this->plugin, $worldName);
			}

			foreach ($texts as $textName => $val) {
				$arr = [
					Data::KEY_X => (float) $val[Data::KEY_OLD_X],
					Data::KEY_Y => (float) $val[Data::KEY_OLD_Y],
					Data::KEY_Z => (float) $val[Data::KEY_OLD_Z],
					Data::KEY_TEXTS => [
						$val[Data::KEY_OLD_TITLE] . "#" . $val[Data::KEY_OLD_TEXT]
					],
				];
				$floatingTextCluster = FloatingTextCluster::fromArray((string)$textName, $arr);
				$floatingTextData->store($floatingTextCluster);
			}

			$floatingTextData->save();
		}
		$this->removeFile();
	}

	private function removeFile() {
		unlink($this->getPath());
	}

}

File: src/main/java/jp/mcbe/fuyutsuki/Texter/data/OldFloatingTextData.java
package jp.mcbe.fuyutsuki.Texter.data;

import jp.mcbe.fuyutsuki.Texter.text.FloatingTextCluster;
import org.powernukkitx.plugin.PluginBase;
import org.powernukkitx.utils.Config;

import java.io.File;
import java.util.ArrayList;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

public final class OldFloatingTextData extends Config {

	public static final String FILE_FT = "ft.json";
	public static final String FILE_UFT = "uft.json";

	private final String filePath;

	private final PluginBase plugin;

	public OldFloatingTextData(PluginBase plugin, String path, String file) {
		super(path + file, Config.JSON);
		this.plugin = plugin;
		this.filePath = path + file;
	}

	@SuppressWarnings("unchecked")
	public void convert() {
		Map<String, Object> fts = this.getAll();

		for (Map.Entry<String, Object> worldEntry : fts.entrySet()) {
			String worldName = worldEntry.getKey();
			Map<String, Object> texts = (Map<String, Object>) worldEntry.getValue();

			FloatingTextData floatingTextData = FloatingTextData.getInstance(worldName);
			if (floatingTextData == null) {
				floatingTextData = new FloatingTextData(plugin, worldName);
			}

			for (Map.Entry<String, Object> textEntry : texts.entrySet()) {
				String textName = textEntry.getKey();
				Map<String, Object> val = (Map<String, Object>) textEntry.getValue();

				Map<String, Object> arr = new LinkedHashMap<>();
				arr.put(Data.KEY_X, ((Number) val.get(Data.KEY_OLD_X)).doubleValue());
				arr.put(Data.KEY_Y, ((Number) val.get(Data.KEY_OLD_Y)).doubleValue());
				arr.put(Data.KEY_Z, ((Number) val.get(Data.KEY_OLD_Z)).doubleValue());

				List<String> texts_ = new ArrayList<>();
				texts_.add(val.get(Data.KEY_OLD_TITLE) + "#" + val.get(Data.KEY_OLD_TEXT));
				arr.put(Data.KEY_TEXTS, texts_);

				FloatingTextCluster floatingTextCluster = FloatingTextCluster.fromMap(textName, arr);
				floatingTextData.store(floatingTextCluster, false);
			}

			floatingTextData.save();
		}

		removeFile();
	}

	private void removeFile() {
		new File(this.filePath).delete();
	}
}